enterprise_gateway/services/external/k8s_client.py

```diff
@@ -0,0 +1,25 @@
+"""Contains factory to create kubernetes api client instances using a single confguration"""
+import os
+
+from kubernetes import client, config
+
+
+class KubernetesClientFactory:
+    """Manages kubernetes client creation from environment variables"""
+    def __init__(self):
```

Comment:
Nit: I would give yourself the facility to log to the console. This way you can validate the `EG_KUBECONFIG_CONTEXT`:

```suggestion
class KubernetesClientFactory:
    """Manages kubernetes client creation from environment variables"""
    
    _kernel_manager : RemoteKernelManager
    _log : Logger
    
    def __init__(self, kernel_manager : RemoteKernelManager):
         self._kernel_manager = kernel_manager
         self._log = kernel_manager.log
```